tests/simplex/step-38.cc

```diff
@@ -0,0 +1,445 @@
+
+    return return_value;
+  }
+
+  template <int dim>
+  class RightHandSide : public Function<dim>
+  {
+  public:
+    virtual double
+    value(const Point<dim> &p, const unsigned int component = 0) const override;
+  };
+
+  template <>
+  double
+  RightHandSide<2>::value(const Point<2> &p,
+                          const unsigned int /*component*/) const
+  {
+    return (-8. * p(0) * p(1));
+  }
+
+  template <>
+  double
+  RightHandSide<3>::value(const Point<3> &p,
+                          const unsigned int /*component*/) const
+  {
+    using numbers::PI;
+
+    Tensor<2, 3> hessian;
+
+    hessian[0][0] = -PI * PI * sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    hessian[1][1] = -PI * PI * sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    hessian[2][2] = sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+
+    hessian[0][1] = -PI * PI * cos(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+    hessian[1][0] = -PI * PI * cos(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+
+    hessian[0][2] = PI * cos(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    hessian[2][0] = PI * cos(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+
+    hessian[1][2] = -PI * sin(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+    hessian[2][1] = -PI * sin(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+
+    Tensor<1, 3> gradient;
+    gradient[0] = PI * cos(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+    gradient[1] = -PI * sin(PI * p(0)) * sin(PI * p(1)) * exp(p(2));
+    gradient[2] = sin(PI * p(0)) * cos(PI * p(1)) * exp(p(2));
+
+    Point<3> normal = p;
+    normal /= p.norm();
+
+    return (-trace(hessian) + 2 * (gradient * normal) +
+            (hessian * normal) * normal);
+  }
+
+  template <int spacedim>
+  LaplaceBeltramiProblem<spacedim>::LaplaceBeltramiProblem(
+    const unsigned degree)
+    : dof_handler(triangulation)
+    , fe(degree)
+#ifdef USE_SIMPLEX
+    // Quadratic mapping for simplex is still not implemented.
+    , mapping(Simplex::FE_P<dim, spacedim>(degree))
+#else
+    , mapping(degree)
+#endif
+  {}
+
+  template <int spacedim>
+  void
+  LaplaceBeltramiProblem<spacedim>::make_grid_and_dofs()
+  {
+    {
+      Triangulation<spacedim> volume_mesh;
+      GridGenerator::half_hyper_ball(volume_mesh);
+
+      std::set<types::boundary_id> boundary_ids;
+      boundary_ids.insert(0);
+
+#ifdef USE_SIMPLEX
+      Triangulation<dim, spacedim> tria_temp;
+      GridGenerator::extract_boundary_mesh(volume_mesh,
+                                           tria_temp,
+                                           boundary_ids);
+      tria_temp.set_all_manifold_ids(0);
+      tria_temp.set_manifold(0, SphericalManifold<dim, spacedim>());
+      GridGenerator::convert_hypercube_to_simplex_mesh(tria_temp,
+                                                       triangulation);
+      for (const auto i : tria_temp.get_manifold_ids())
+        if (i != numbers::flat_manifold_id)
+          triangulation.set_manifold(i, tria_temp.get_manifold(i));
```

Comment:
I am wondering if the following also works:
```suggestion
      Triangulation<spacedim> volume_mesh_temp;
      GridGenerator::convert_hypercube_to_simplex_mesh(volume_mesh, volume_mesh_temp);
      GridGenerator::extract_boundary_mesh(volume_mesh,
                                           triangulation,
                                           boundary_ids);
      triangulation.set_all_manifold_ids(0);
      triangulation.set_manifold(0, SphericalManifold<dim, spacedim>());
```

Reply:
This should not work because the routine `GridGenerator::extract_boundary_mesh` currently relies on  'GeometryInfo' and does not work for simplex meshes in 3D. I can think about how to remove `GeometryInfo` from this routine (maybe in another PR).

Reply:
>  I can think about how to remove GeometryInfo from this routine (maybe in another PR).

That would be great!